### PR TITLE
autcomplete select

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[GdO-Speziessuche auf ngx-mat-select-search umstellen](https://trello.com/c/b6wJ4w2w/604-gdo-speziessuche-auf-ngx-mat-select-search-umstellen)

File: projects/wave-core-new/src/lib/util/directives/autocomplete-select.directive.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+@Directive({
+    selector: '[waveAutocompleteSelect]',
+    exportAs: 'waveAutocompleteSelect',
+})
+export class AutocompleteSelectDirective<T> implements OnChanges, OnDestroy {
+    @Input('waveAutocompleteSelectAllValues') allValues: Array<T> = [];
+    @Input('waveAutocompleteSelectSearchPredicate') searchPredicate: AutocompleteSelectPredicateFunction<T> = DEFAULT_PREDICATE_FN;
+
+    readonly filterFormControl = new FormControl();
+    readonly _filteredValues = new ReplaySubject<Array<T>>();
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/util/directives/autocomplete-select.directive.ts:14]: why is this prefixed with _ ?

Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/util/directives/autocomplete-select.directive.ts:14]: filteredValues is also something that is outputtet via a get-method, but only as an observable and not as a changeable subject.

**Review by jdroenner — APPROVED** [review]

**Comment by jdroenner** [comment]
bors merge

**Pull request merged**